pkg/api/datasources.go

```diff
@@ -687,11 +688,7 @@ func (hs *HTTPServer) convertModelToDtos(ctx context.Context, ds *datasources.Da
 
 	secrets, err := hs.DataSourcesService.DecryptedValues(ctx, ds)
 	if err == nil {
-		for k, v := range secrets {
-			if len(v) > 0 {
-				dto.SecureJsonFields[k] = true
-			}
-		}
+		dto.SecureJsonFields = pluginsettings.ToSecureJsonFields(secrets)
```

Comment:
This is a bit strange for the `datasources` package to use `pluginsettings` to calculate the json fields. Same logic, but different models. I've pushed a fix to my branch that should do the trick